GVFS/GVFS.FunctionalTests/FileSystemRunners/BashRunner.cs

```diff
@@ -86,15 +94,22 @@ public static void DeleteDirectoryWithUnlimitedRetries(string path)
             }
         }
 
-        public override bool FileExists(string path)
+        public bool IsSymbolicLink(string path)

```

Comment:
> This would make it so the test didn't have to use the BashRunner directly.

My thought had been that symbolic link tests would just always use BashRunner (much like how many of the tests explicitly use `SystemIORunner`). 

BashRunner natively supports checking if a path is a symlink (and it's cross platform) which made it a good candidate for the symlink tests.

I'm not sure that running these tests with additional runners gets us much more coverage.  

What do you think about this approach?

Reply:
That should be fine for now but if users can create symlinks in other ways like System.IO or cmd we should probably be testing those to make sure git and VFSforGit are handling them correctly.  Maybe all the runners are hitting the same code at some point so that would make it easier to say they are testing the same thing?

Reply:
> That should be fine for now but if users can create symlinks in other ways like System.IO or cmd we should probably be testing those to make sure git and VFSforGit are handling them correctly.

Yep I agree.  When we add code to support the detection of user-created symlinks we'll want to make sure that all of our runners can create them .

Currently the tests are just validating that VFS for Git is writing symlinks when appropriate, and so I think it's less critical for these tests to check with every runner.